framework/include/meshgenerators/MeshGenerator.h

```diff
@@ -68,6 +68,92 @@ class MeshGenerator : public MooseObject, public MeshMetaDataInterface
   template <typename T>
   T & declareMeshProperty(const std::string & data_name, const T & init_value);
 
+  /// This collects all the type names that are declared as Mesh Metadata
+  const std::vector<std::string> _type_names = {
+      "std::vector<unsigned int>",
+      "double",
+      "std::vector<double>",
+      "unsigned short",
+      "unsigned int",
+      "bool",
+      "unsigned long long",
+      "std::string",
+      "int",
+      "libMesh::Point",
+      "std::vector<unsigned int>",
+      "std::vector<int>",
+      "std::vector<unsigned short>",
+      "std::vector<unsigned long long>",
+      "std::vector<libMesh::Point>",
+      "std::vector<std::vector<double>>",
+      "std::map<std::string, std::pair<unsigned short, unsigned long long>, "
+      "std::less<std::string>, std::allocator<std::pair<std::stringconst, std::pair<unsigned "
+      "short, unsigned long long> > > >",
+      "short",
+      "std::map<unsigned short, std::vector<std::vector<unsigned short>>, std::less<unsigned "
+      "short>, std::allocator<std::pair<unsigned short const, std::vector<std::vector<unsigned "
+      "short>> > > >",
+      "std::map<unsigned short, std::vector<std::vector<std::string>>, std::less<unsigned short>, "
+      "std::allocator<std::pair<unsigned short const, std::vector<std::vector<std::string>> > > >",
+      "std::vector<std::string>",
+      "std::vector<std::vector<unsigned short>>",
+      "std::vector<std::vector<std::string>>",
+      "std::map<short, libMesh::VectorValue<double>, std::less<short>, "
+      "std::allocator<std::pair<short const, libMesh::VectorValue<double> > > >"};
```

Comment:
looping through the map, can we build this list automatically?

Reply:
That's a good idea.

Reply:
@GiudGiud I managed to use a string-enum map to do the work instead of a manual mapping process.

I thought about your question here. The metadata map is created in MG constructors. So it only contains metadata that are used in the specific input file. However, we want a list here that covers everything and it will be used during compilation (as we are still "manually" giving the typenames).